### PR TITLE
[batch][dag2] Switch to Python Modules

### PR DESCRIPTION
**Pull request by danking**
Future enhancements to batch will necessitate the proper use of modules.

### CODE REVIEW

File: batch/test-locally.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 set -ex
 
+. activate hail-batch
```
Comment (jigold) [inline batch/test-locally.sh]: What does this do?

File: batch/Dockerfile.test [hunk 2]
```diff
@@ -8,6 +8,7 @@ RUN pip install kubernetes
 RUN pip install cerberus
 
 COPY batch /batch
+RUN pip install /batch
```
Comment (jigold) [inline batch/Dockerfile.test]: I didn't know you could install a local python module.

File: batch/test-locally.sh (hunk 1 above)
Comment (danking) [reply batch/test-locally.sh]: [POSIX `. foo args ...`](http://pubs.opengroup.org/onlinepubs/009696699/utilities/dot.html) executes the script `foo` with the arguments `args ...` in the current environment, so if `foo` contains a line:
```bash
VARIABLE=value!
```
then the shell that called `. foo args` will have `VARIABLE` set to `value!`. `./foo` and `bash foo` do not have the same effect.

Practically speaking, it enables the hail-batch conda environment with all the binaries and packages described by `batch/environment.yml`. It's the POSIX-compliant way to say `source activate hail-batch`.

File: batch/Dockerfile.test (hunk 2 above)
Comment (danking) [reply batch/Dockerfile.test]: Neither did I!

Comment (akotlar) [reply batch/Dockerfile.test]: neat use!

**Pull request merged**